### PR TITLE
Limit get owned object size

### PR DESCRIPTION
**Pull request by lxfind**
Add a cap on the number of objects to return. Also make it flexible to eventually support pagination.

### CODE REVIEW

File: crates/sui-storage/src/indexes.rs [hunk 1]
```diff
@@ -565,22 +568,32 @@ impl IndexStore {
...
+        let mut i = 0;
         Ok(self
             .tables
             .owner_index
             .iter()
             // The object id 0 is the smallest possible
-            .skip_to(&(owner, ObjectID::ZERO))?
-            .take_while(move |((object_owner, _), _)| (object_owner == &owner))
+            .skip_to(&(owner, starting_object_id))?
+            .take_while(move |((object_owner, _), _)| {
```
Comment (patrickkuo) [inline crates/sui-storage/src/indexes.rs:593]: we can use `.take(count)` instead of using a counter

**Review by patrickkuo — APPROVED** [review]
just a small nit on using Iterator::take instead of manually counting the return item.

**Comment by Jordan-Mysten** [comment]
@lxfind Just a heads up, this will break some functionality in the wallet if users have a lot of objects in their account.

**Comment by lxfind** [comment]
Happy to make the size limit larger, but there will need to be some limit, though.

**Pull request merged**